src/structurizr/model/implied_relationship_strategies.py

```diff
@@ -0,0 +1,108 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+Implement various strategies for adding implied relationships to the model.
+
+Implied relationship strategies are used to add relationships to parents when a
+relationship is added to their children.  For example, assuming systems A and B with
+child containers A1 and B1 respectively, then saying that A1 uses B1 implies that
+A also uses B (and A uses B1 and A1 uses B).  Each strategy is represented as a
+function that can be set on `Model.implied_relationship_strategy`, with the default
+being to do nothing.
+"""
+
+from itertools import product
+from typing import List
+
+from .element import Element
+from .relationship import Relationship
+from .software_system import SoftwareSystem
+
+
+def default_implied_relationship_strategy(relationship: Relationship):
```

Comment:
Maybe call this one 
```suggestion
def ignore_implied_relationships_strategy(relationship: Relationship):
```

Reply:
I was mirroring Java, but agree that "ignore" is more descriptive.